datafusion/src/physical_plan/hash_aggregate.rs

```diff
@@ -368,7 +376,10 @@ fn group_aggregate_batch(
             group_values
                 .iter()
                 .zip(group_state.group_by_values.iter())
-                .all(|(array, scalar)| scalar.eq_array(array, row))
+                .zip(null_information.iter())
+                .all(|((array, scalar), has_nulls)| {
+                    scalar.eq_array(array, row, has_nulls)
+                })
```

Comment:
I think broadly speaking this is the pattern, though I am not sure how well the compiler will be able to optimize given a has nulls parameter `has_nulls` -- it might need to be hoisted out of this loop (the call to `all` here)

Have you had a chance to try profiling?

Reply:
Hi I don't understand the suggestion about hoisting `has_nulls` out of the `all` 

Would you be able to please sketch it out in pseudo code or similar? 

I haven't used macros much and will go educate myself on how they work. Will also get to profiling (although maybe not until the weekend).